src/lib/io/sol-iio.c

```diff
@@ -654,6 +656,28 @@ channel_get_pure_name(const char *name)
 }
 
 static bool
+iio_set_oversampling_ratio(struct sol_iio_device *device, const struct sol_iio_config *config)
+{
+    char path[PATH_MAX];
+    struct sol_str_table *iter;
+
+    SOL_NULL_CHECK(device, false);
```

Comment:
YES,  check the input parameter pointer in function "iio set_sampling_frequency" too.